client/tracing/src/lib.rs

```diff
@@ -259,29 +256,22 @@ fn parse_target(s: &str) -> (String, Level) {
 	}
 }
 
-impl Subscriber for ProfilingSubscriber {
-	fn enabled(&self, metadata: &Metadata<'_>) -> bool {
-		if metadata.target() == PROXY_TARGET || self.check_target(metadata.target(), metadata.level()) {
-			log::debug!(target: "tracing", "Enabled target: {}, level: {}", metadata.target(), metadata.level());
-			true
-		} else {
-			log::debug!(target: "tracing", "Disabled target: {}, level: {}", metadata.target(), metadata.level());
-			false
-		}
+impl<S: Subscriber> Layer<S> for ProfilingLayer {
+	fn enabled(&self, metadata: &Metadata<'_>, _ctx: Context<S>) -> bool {
```

Comment:
The way this is implemented means that this layer will filter globally (inc from the FmtSubscriber), so I think we need to leave it at default, which is always `true`.

Then we need to do the filtering in `new_span` and keep corresponding noop in `on_enter` etc if span `Id` is not found.

Also somehow we need to enable the callsites required for profiling in the FmtSubscriber.